### PR TITLE
MDL-78806 better page title documentation

### PR DESCRIPTION
**Pull request by junpataleta**
Dev docs for when MDL-78806 gets integrated

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/729"></a>

### CODE REVIEW

File: general/development/policies/accessibility.md [hunk 1]
```diff
@@ -121,7 +123,48 @@ See [the ARIA best practice advice on landmarks](https://www.w3.org/TR/wai-aria-
...
+
+- The page title must be accurate and informative.
+- If the page causes a change of context (e.g. a search functionality), it should describe the result or change of context to the user.
+- It should be concise.
+- If possible, it should uniquely identify the page.
+- The most identifying information should come first.
+
+#### Example
+
+Consider that a student in the submission page of an assignment called `Kinetics problem set 1` in the `Physics 101` in the `Moodle University`.
```
Comment (andrewnicols) [inline general/development/policies/accessibility.md:136]:
```suggestion
Consider that a student is on the submission page of an assignment actvity called `Kinetics problem set 1` in the `Physics 101` course in the `Wattle Grove University` Moodle site.
```

Better not to use "Moodle" in the name of the site to improve clarity of the sentence.

File: general/development/policies/accessibility.md [hunk 2]
```diff
@@ -121,7 +123,48 @@ See [the ARIA best practice advice on landmarks](https://www.w3.org/TR/wai-aria-
...
+
+Then a suitable page title for the page would be something like:
+
+<ValidExample>
+
+The most unique identifying information first represented by the activity name and its sub-page, then followed by broader identifiers such as the course name and the site name.
+
+`Kinetics problem set 1: Submit assignment | Physics 101 | Moodle University`
```
Comment (andrewnicols) [inline general/development/policies/accessibility.md:144]:
```suggestion
`Kinetics problem set 1: Submit assignment | Physics 101 | Wattle Grove University`
```

File: general/development/policies/accessibility.md [hunk 3]
```diff
@@ -121,7 +123,48 @@ See [the ARIA best practice advice on landmarks](https://www.w3.org/TR/wai-aria-
...
+
+</ValidExample>
+
+<ValidExample>
+
+The most unique identifying information first represented by sub-page's name followed by the activity name that the page belongs to, then followed by broader identifiers such as the course name and the site name.
```
Comment (andrewnicols) [inline general/development/policies/accessibility.md:150]:
```suggestion
The most unique identifying information first represented by the name of the sub-page, followed by the activity name that the page belongs to, then followed by broader identifiers such as the course name and the site name.
```

File: general/development/policies/accessibility.md [hunk 4]
```diff
@@ -121,7 +123,48 @@ See [the ARIA best practice advice on landmarks](https://www.w3.org/TR/wai-aria-
...
+
+`Submit assignment | Kinetics problem set 1 | Physics 101 | Moodle University`
+
+</ValidExample>
+
+#### Separating components of a page title
+
+When separating the components of the page tile, please use the `moodle_page:TITLE_SEPARATOR` constant.
```
Comment (andrewnicols) [inline general/development/policies/accessibility.md:158]: Please provide an example of how you may do so?

File: general/development/policies/accessibility.md [hunk 5]
```diff
@@ -121,7 +123,48 @@ See [the ARIA best practice advice on landmarks](https://www.w3.org/TR/wai-aria-
 
 All pages should have a unique title that describes the current page.
 
-See [the WCAG 2.1 success criteria for web page titles](https://www.w3.org/TR/WCAG21/#page-titled) for further information.
+Some tips for providing a meaningful page title:
+
+- The page title must be accurate and informative.
+- If the page causes a change of context (e.g. a search functionality), it should describe the result or change of context to the user.
```
[the remark this answers is not in the document]
Comment (andrewnicols) [reply general/development/policies/accessibility.md:129]: Note: Most style guides recommend _against_ use of `e.g.` and `i.e.`:
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg
https://developers.google.com/style/abbreviations#dont-use

File: general/development/policies/accessibility.md [hunk 6]
```diff
@@ -121,7 +123,48 @@ See [the ARIA best practice advice on landmarks](https://www.w3.org/TR/wai-aria-
...
+
+</ValidExample>
+
+#### Separating components of a page title
+
+When separating the components of the page tile, please use the `moodle_page:TITLE_SEPARATOR` constant.
+
+#### Site name on the page title
+
+There's no need to add the site's name when setting the page title using `$PAGE->set_title()`. The site name is automatically appended at the end of the page title by default by `$PAGE->set_title()`. The displayed site name on the page title will depend on the admin setting `$CFG->sitenameintitle` whether it's set to show the site's full name or the site's short name.
```
Comment (andrewnicols) [inline general/development/policies/accessibility.md:162]: I'd suggest noting this in an admonition to make it clearer why, and the impacts and options.

```suggestion
You should not add the name of the site when setting the page title using `$PAGE->set_title()`. The site name is automatically appended to the end of the page title in the correct format when using `$PAGE->set_title()`.

:::info

Administrators can use the `sitenameinititle` configuration setting to configure how this is shown in the title with possible options including:

- the _full name_ of the site, for example "Wattle Grove University"
- the _short name_ of the site, for example: "WGU"

THis is automatically handled by `$PAGE->set_title()`.
:::
```

**Review by andrewnicols — CHANGES_REQUESTED** [review]

File: general/development/policies/accessibility.md (hunk 4 above)
Comment (junpataleta) [reply general/development/policies/accessibility.md]: [Example code](https://github.com/moodle/devdocs/pull/729/files#diff-df8f1f100a26a2d67a6012a3b0fe8884c9f6a095acd0b9e0ba1d817104912c69R168-R180) added

**Review by andrewnicols — APPROVED** [review]
Approving but I think we need to be careful and re-assess the example:
```
// Activity name and its sub-page as the unique identifying information.
$pagename = format_string($cm->name) . ': ' . get_string('view');
```

This is a great example of lang string concatenation, which we should avoid where possible.

**Pull request merged**